include/triton/Dialect/TritonGPU/IR/TritonGPUOps.td

```diff
@@ -239,35 +159,63 @@ def TTG_InsertSliceAsyncOp : TTG_Op<"insert_slice_async",
     }
   }];
 
-  let hasCustomAssemblyFormat = 1;
+  let assemblyFormat = [{
+    $src `,` $dst (`mask` $mask^)? (`other` $other^)? attr-dict `:` type($src) `->` type($dst)
+  }];
 }
 
-def TTG_AllocTensorOp : TTG_Op<"alloc_tensor", [MemoryEffects<[MemAlloc<SharedMemory>]>,  // Allocate shared memory
-                                                ResultsAreSharedEncoding]> {
+
+// Allocate shared memory
+def TTG_LocalAllocOp : TTG_Op<"local_alloc", [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
   let summary = "allocate tensor";
+  let description = [{
+    This operation allocates buffer in shared memory and return a descriptor
+    containing the address and a view of the buffer.
+  }];
+  let arguments = (ins Optional<TT_Tensor>:$init);
+
+  let assemblyFormat = [{$init attr-dict `:` functional-type(operands, results)}];
+
+  let results = (outs TT_MemDescType:$result);
+}
+
+// Deallocate shared memory
+def TTG_DeallocOp : TTG_Op<"dealloc", [MemoryEffects<[MemFree<SharedMemory>]>]> {
+  let summary = "dealloc buffer";
 
   let description = [{
-    This operation defines a tensor of a particular shape.
-    The contents of the tensor are supposed to be in shared memory.
+    This operation deallocate a buffer explicitly. Using the buffer after this operation is undefined.
   }];
 
-  let assemblyFormat = [{attr-dict `:` type($result)}];
+  let arguments = (ins TT_MemDescType:$ptr);
 
-  let results = (outs TT_Tensor:$result);
+  let assemblyFormat = [{$ptr attr-dict `:` qualified(type($ptr))}];
 }
 
-def TTG_DeallocTensorOp : TTG_Op<"dealloc_tensor", [MemoryEffects<[MemFree<SharedMemory>]>,  // Deallocate shared memory
-                                                    OperandsAreSharedEncoding]> {
-  let summary = "dealloc tensor";
+def TTG_SubviewOp : TTG_Op<"subview", [Pure]> {
+  let summary = "take a subview of the descriptor.";
 
   let description = [{
-    This operation deallocate a tensor explicitly.
-    The contents of the tensor are supposed to be in shared memory.
+    This operation returns a new descriptor representing a subview of the buffer.
+    It doesn't affect the underlying memory. The subview can be rank-reduced.
   }];
+  let arguments = (
+    ins TT_MemDescType:$desc, Variadic<I32>:$offsets);
+
+  let assemblyFormat = [{$desc `[` $offsets `]` attr-dict `:` qualified(type($desc)) `->` qualified(type($result))}];
+
+  let results = (outs TT_MemDescType:$result);
+}
 
-  let arguments = (ins TT_Tensor:$ptr);
+def TTG_LocalLoadOp : TTG_Op<"local_load", [MemoryEffects<[MemRead<SharedMemory>]>]> {
```

Comment:
Wondering whether we should follow the `AsyncCopyToLocalOp` naming, and have "to/from" in the name, like `LoadFromLocalOp` ?

Reply:
That feels a bit verbose. I picked `AsyncCopyToLocalOp` because I wanted to make it clear there if read from global. Maybe I should rename it to `AsyncLoadToLocalOp`?

Reply:
I have avoided commenting on the names, but are we having the naming discussion now?  I have feelpinions....

Reply:
ok, let's see if we can settle on a set of names in a short discussion otherwise I suggest we keep this for later.

Reply:
name changed based on offline discussion